systemtest/src/test/java/io/strimzi/systemtest/KafkaST.java

```diff
@@ -2128,6 +2137,344 @@ void testKafkaOffsetsReplicationFactorHigherThanReplicas() {
+        assertThat(StatefulSetUtils.ssHasRolled(kafkaStatefulSetName(CLUSTER_NAME), kafkaPods), is(false));
+
+        LOGGER.info("Verify values after update");
+        kafkaConfiguration = kubeClient().getConfigMap(KafkaResources.kafkaMetricsAndLogConfigMapName(CLUSTER_NAME)).getData().get("server.config");
+        assertThat(kafkaConfiguration, containsString("offsets.topic.replication.factor=1"));
+        assertThat(kafkaConfiguration, containsString("transaction.state.log.replication.factor=1"));
+        assertThat(kafkaConfiguration, containsString("default.replication.factor=1"));
+
+        kafkaConfigurationFromPod = cmdKubeClient().execInPod(KafkaResources.kafkaPodName(CLUSTER_NAME, 0), "/bin/bash", "-c", "bin/kafka-configs.sh --bootstrap-server localhost:9092 --entity-type brokers --entity-name 0 --describe").out();
+        assertThat(kafkaConfigurationFromPod, containsString("unclean.leader.election.enable=true"));
+
+        InlineLogging il = new InlineLoggingBuilder().withLoggers(Collections.singletonMap("kafka.logger.level", "INFO")).build();
+
+        Map<String, String> kafkaPodsSnapshot = StatefulSetUtils.ssSnapshot(kafkaStatefulSetName(CLUSTER_NAME));
+
+        LOGGER.info("Updating logging of Kafka cluster");
+        KafkaResource.replaceKafkaResource(CLUSTER_NAME, k -> {
+            KafkaClusterSpec kafkaClusterSpec = k.getSpec().getKafka();
+            kafkaClusterSpec.setLogging(il);
+        });
+
+        StatefulSetUtils.waitTillSsHasRolled(kafkaStatefulSetName(CLUSTER_NAME), kafkaReplicas, kafkaPodsSnapshot);
+
+    }
+
+    @Test
+    void testDynamicConfigurationWintExternalListeners() {
+        int kafkaReplicas = 2;
+        int zkReplicas = 1;
+        Map<String, Object> kafkaConfig = new HashMap<>();
+        kafkaConfig.put("offsets.topic.replication.factor", "1");
+        kafkaConfig.put("transaction.state.log.replication.factor", "1");
+        kafkaConfig.put("default.replication.factor", "1");
+        kafkaConfig.put("log.message.format.version", "2.4");
+
+        Map<String, Object> updatedKafkaConfig = new HashMap<>();
+        updatedKafkaConfig.put("offsets.topic.replication.factor", "1");
+        updatedKafkaConfig.put("transaction.state.log.replication.factor", "1");
+        updatedKafkaConfig.put("default.replication.factor", "1");
+        updatedKafkaConfig.put("log.message.format.version", "2.4");
+        updatedKafkaConfig.put("unclean.leader.election.enable", "true");
+
+        KafkaResource.kafkaEphemeral(CLUSTER_NAME, kafkaReplicas, zkReplicas)
+                .editSpec()
+                .editKafka()
+                    .withNewListeners()
+                        .withNewKafkaListenerExternalLoadBalancer()
+                        .endKafkaListenerExternalLoadBalancer()
+                        .withNewPlain()
+                        .endPlain()
```

Comment:
This is not needed for the case you are testing :). I do not see any verification method for this plain listener or some exchange of messages.

Reply:
Even plain listener adds values to the `advertised.listeners`. I was thinking it would be good to test is together with other changes. You are right it would be better to remove it.